session.go

```diff
@@ -546,15 +551,20 @@ func (addr *ServerAddr) String() string {
 
 // TCPAddr returns the resolved TCP address for the server.
 func (addr *ServerAddr) TCPAddr() *net.TCPAddr {
-	return addr.tcp
+	return (*net.TCPAddr)(addr.addr.(*net.TCPAddr))
```

Comment:
`return addr.addr.(*net.TCPAddr)`
Please do add a comment saying that this will panic if the ServerAddr is a socket address. Or maybe do 
```
tcp, ok := addr.addr.(*net.TCPAddr)
if !ok{
return nil
}
return tcp
```
And document the behaviour.